numpyro/distributions/continuous.py

```diff
@@ -651,6 +651,151 @@ def variance(self):
+
+        loc, cov_factor, cov_diag = promote_shapes(loc[..., np.newaxis], cov_factor, cov_diag[..., np.newaxis])
+        batch_shape = lax.broadcast_shapes(np.shape(loc), np.shape(cov_factor), np.shape(cov_diag))[:-2]
+        self.loc = np.broadcast_to(loc[..., 0], batch_shape + event_shape)
+        self.cov_factor = cov_factor
+        self.cov_diag = cov_diag
+        self._capacitance_tril = _batch_capacitance_tril(cov_factor, cov_diag)
+        super(LowRankMultivariateNormal, self).__init__(batch_shape = batch_shape, event_shape = event_shape, validate_args=validate_args)
+
+    @property
+    def mean(self):
+        return self.loc
+
+    @lazy_property
+    def variance(self):
+        raw_variance = np.square(self.cov_factor).sum(-1) + self.cov_diag
+        return np.broadcast_to(raw_variance, self.batch_shape + self.event_shape)
+
+    @lazy_property
+    def scale_tril(self):
+        # The following identity is used to increase the numerically computation stability
+        # for Cholesky decomposition (see http://www.gaussianprocess.org/gpml/, Section 3.4.3):
+        #     W @ W.T + D = D1/2 @ (I + D-1/2 @ W @ W.T @ D-1/2) @ D1/2
+        # The matrix "I + D-1/2 @ W @ W.T @ D-1/2" has eigenvalues bounded from below by 1,
+        # hence it is well-conditioned and safe to take Cholesky decomposition.
+        cov_diag_sqrt_unsqueeze = np.expand_dims(np.sqrt(self.cov_diag), axis=-1)
+        Dinvsqrt_W = self.cov_factor / cov_diag_sqrt_unsqueeze
+        K = np.matmul(Dinvsqrt_W, np.swapaxes(Dinvsqrt_W, -1, -2))
+        K = np.add(K, np.identity(K.shape(-1)))
+        scale_tril = cov_diag_sqrt_unsqueeze * np.linalg.cholesky(K)
+        return scale_tril
+
+    @lazy_property
+    def covariance_matrix(self):
+        covariance_matrix = self.cov_diag + np.matmul(
+            self.cov_factor, np.swapaxes(self.cov_factor, -1, -2)
+            )
+        return covariance_matrix
+
+    @lazy_property
+    def precision_matrix(self):
+        # We use "Woodbury matrix identity" to take advantage of low rank form::
+        #     inv(W @ W.T + D) = inv(D) - inv(D) @ W @ inv(C) @ W.T @ inv(D)
+        # where :math:`C` is the capacitance matrix.
+        Wt_Dinv = (np.swapaxes(self.cov_factor, -1, -2)
+                   / np.expand_dims(self.cov_diag, axis=-2))
+        A = solve_triangular(Wt_Dinv, self._capacitance_tril, lower=True)
+        return np.reciprocal(self.cov_diag) - np.matmul(np.swapaxes(A, -1, -2), A)
+
+    def sample(self, key, sample_shape=()):
+        W_shape = sample_shape + self.cov_factor.shape[-1:]
+        eps_W = random.normal(key, W_shape)
+        eps_D = random.normal(key, sample_shape)
+        return (self.loc + _batch_mv(self.cov_factor, eps_W)
+                + np.sqrt(self.cov_diag) * eps_D)
+
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
```

Comment:
nit: it is better to use
```
@validate_sample
def log_prob(self, value):
    diff = ...
```
so the output log_prob is correctly masked for out-of-support samples.

Reply:
@TuanNguyen27 You don't need those two lines `if self._validate...` when we use the decorator `validate_sample`.